File: benchmarks/tsb/bench_series_to_string.ts
/**
 * Benchmark: seriesToString on 1k-element Series
 */
import { Series, seriesToString } from "../../src/index.js";

const N = 1_000;
const WARMUP = 3;
const ITERATIONS = 10;
const data = Array.from({ length: N }, (_, i) => i * 0.1);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) seriesToString(s);
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) seriesToString(s);
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "series_to_string",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_series_to_string.py
"""Benchmark: Series.to_string on 1k-element pandas Series"""
import json, time
import pandas as pd

N = 1_000
WARMUP = 3
ITERATIONS = 10
s = pd.Series([i * 0.1 for i in range(N)])

for _ in range(WARMUP):
    s.to_string()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.to_string()
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "series_to_string", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
